src/utils/network_calculations/ybus_calculations.jl

```diff
@@ -243,12 +246,52 @@ function _goderya(ybus::SparseArrays.SparseMatrixCSC)
     return I
 end
 
-function validate_connectivity(sys::System)
-    a = Adjacency(sys; check_connectivity = false)
-    return validate_connectivity(a.data, collect(get_components(Bus, sys)), a.lookup[1])
+function validate_connectivity(sys::System; method::Union{Nothing, Function} = nothing)
+    buses = sort!(
+        collect(get_components(Bus, sys, x -> get_bustype(x) != BusTypes.ISOLATED)),
+        by = b -> get_number(b),
+    )
+    branches = get_components(Branch, sys, get_available)
+
+    a = Adjacency(branches, buses; check_connectivity = false)
+
+    return validate_connectivity(a.data, buses, a.lookup[1]; method = method)
+end
+
+function validate_connectivity(
+    M,
+    nodes,
+    bus_lookup;
+    method::Union{Nothing, Function} = nothing,
+)
+    preferred_method =
+        Base.summarysize(M) > 200_000 ? calc_connected_components : goderya_connectivity
+
+    if method === nothing
+        method = preferred_method
+    end
+
+    if method == goderya_connectivity
+        connected = goderya_connectivity(M, nodes, bus_lookup)
+    elseif method == calc_connected_components
+        @info "Validating connectivity with network traversal"
+        cc = calc_connected_components(M, bus_lookup)
+        if length(cc) != 1
+            @warn "Network has at least $(length(cc)) connected components with $(length.(cc)) nodes"
+            connected = false
+        else
+            connected = true
+        end
+    end
+
+    return connected
 end
 
-function validate_connectivity(M, nodes, bus_lookup)
+function goderya_connectivity(M, nodes, bus_lookup)
+    @info "Validating connectivity with Goderya algorithm"
+    Base.summarysize(M) > 200_000 &&
```

Comment:
We should limit to 15k buses the use of this method. If we have a full matrix with Float64 it will take 4gb of memory. Since most users would start with a commercial laptop we should limit the memory footprint to 4 gb. 

Goderya will tend to generate full matrices from the M * M operation.